sklearn/feature_selection/_rfe.py

```diff
@@ -729,9 +832,19 @@ def fit(self, X, y, groups=None):
             multi_output=True,
         )
 
+        if _routing_enabled():
+            params.update({"groups": groups})
```

Comment:
we should only do this if `groups is not None`

right now all splitters accept `groups` but ignore them, but with slep6 in place, that can change.